### PR TITLE
Generalize 1817 token reduction code and add to 1867

### PR DESCRIPTION
**Pull request by jenf**
Also fix repaying of loans for 1867

### CODE REVIEW

File: lib/engine/step/token_merger.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        others.flat_map(&:tokens)
+      end
+
+      def remove_duplicate_tokens(surviving, others)
+        # If there are 2 station markers on the same city the
+        # surviving company must remove one and place it on its charter.
+        # In the case of NY tiles this is ambigious and must be solved by the user
+
+        others = others_tokens(others).map(&:city).compact
+        surviving.tokens.each do |token|
```
Comment (tobymao) [inline lib/engine/step/token_merger.rb:28]: do you need to do a dup here? dose remove! mutate tokens?

**Review by tobymao — APPROVED** [review]

File: lib/engine/step/token_merger.rb (hunk 1 above)
Comment (jenf) [reply lib/engine/step/token_merger.rb:28]: huh?

Comment (jenf) [reply lib/engine/step/token_merger.rb:28]: what dup? Sorry unsure what you mean

Comment (tobymao) [reply lib/engine/step/token_merger.rb:28]: surviving.tokens.dup.each do 

because you do a remove right?

Comment (jenf) [reply lib/engine/step/token_merger.rb:28]: oh, i see what you mean. no the token.remove! just un-places the token. (This is exactly the same as the original 1817 code)

**Pull request merged**